aws/resource_aws_worklink_fleet.go

```diff
@@ -0,0 +1,399 @@
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateWorklinkFleetName,
+			},
+			"display_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(0, 100),
+			},
+			"audit_stream_arn": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateArn,
+			},
+			"network": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"vpc_id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"security_group_ids": {
+							Type:     schema.TypeSet,
+							Required: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+						},
+						"subnet_ids": {
+							Type:     schema.TypeSet,
+							Required: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+						},
+					},
+				},
+			},
+			"device_ca_certificate": {
+				Type:     schema.TypeString,
+				Optional: true,
+				StateFunc: func(v interface{}) string {
+					s, ok := v.(string)
+					if !ok {
+						return ""
+					}
+					return strings.TrimSpace(s)
+				},
+			},
+			"identity_provider": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"saml_metadata": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringLenBetween(1, 204800),
+						},
+					},
+				},
+			},
+			"company_code": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_updated_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"optimize_for_end_user_location": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+		},
+	}
+}
+
+func resourceAwsWorkLinkFleetCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).worklinkconn
+
+	input := &worklink.CreateFleetInput{
+		FleetName:                  aws.String(d.Get("name").(string)),
+		OptimizeForEndUserLocation: aws.Bool(d.Get("optimize_for_end_user_location").(bool)),
+	}
+
+	if v, ok := d.GetOk("display_name"); ok {
+		input.DisplayName = aws.String(v.(string))
+	}
+
+	resp, err := conn.CreateFleet(input)
+	if err != nil {
+		return fmt.Errorf("Error creating Worklink Fleet: %s", err)
+	}
+
+	d.SetId(aws.StringValue(resp.FleetArn))
+	return resourceAwsWorkLinkFleetUpdate(d, meta)
```

Comment:
We generally prefer to avoid calling Update after Create since it can cause maintenance issues long-term instead this should call Read and implement any necessary logic to properly create the resource above.

In this case, this resource would currently always call `UpdateFleetMetadata` after creation as `optimize_for_end_user_location` will have the difference `"" => "false/true"` and `display_name` will have a difference if it is not empty. The `UpdateFleetMetadata` call needs to be protected with a `!d.IsNewResource()` conditional if we are keeping Update after Create.

Reply:
Thanks.
I remove calling update method in creating step.